src/image/pixels.js

```diff
@@ -469,8 +469,19 @@ p5.prototype._copyHelper = (
  * image of a brickwall
  * image of a brickwall with less detail
  */
+
+/**
+ * @method filter
+ * @param {p5.Shader}  shaderFilter  A shader that's been loaded
+ */
 p5.prototype.filter = function(operation, value) {
   p5._validateParameters('filter', arguments);
+
+  if (this._renderer.isP3D) {
```

Comment:
I think we should continue this discussion about whether or not webGL filter should attempt to be used even if the source sketch was not in webGL model

Reply:
Agreed, I think that's the primary value of this whole project, speeding up existing `filter()` calls in P2D sketches. So P2D sketches would have to create a hidden webGL context if `filter` is used.

I think shader filter usage should be automatic while including an opt-out. Something like `filter(CONSTANT, useWebGL=false)`?

And I think that should be included in the next PR that implements the webGL shaders for filter constants.

Reply:
I agree about the opt-out, most people would benefit from the hidden webGL context, but it feels like there would be some negative side-effects eventually if there was no way to avoid that. Am I correct in thinking that there might be situations where a shader heavy project might benefit from strategically shifting some effects to the CPU?

Reply:
I think we can check if webgl support is available and then attempt to filter on the GPU for the filter constants. Agree that it would be good to opt out if possible but it does add some slight code complexity here